identity-server/src/IdentityServer/Validation/Default/PrivateKeyJwtSecretValidator.cs

```diff
@@ -63,7 +63,15 @@ public async Task<SecretValidationResult> ValidateAsync(IEnumerable<Secret> secr
         var fail = new SecretValidationResult { Success = false };
         var success = new SecretValidationResult { Success = true };
 
-        if (parsedSecret.Type != ParsedSecretTypes.JwtBearer)
+        // Decide whether to enforce strict audience validation or not.
+        var enforceStrictAud = _options.Preview.StrictClientAssertionAudienceValidation;
+
+        if (enforceStrictAud && parsedSecret.Type != "client-authentication+jwt")
```

Comment:
Should the string `client-authentication+jwt` be put into a constant since it's used in two locations?